src/pages/profile/index.tsx

```diff
@@ -6,16 +6,46 @@ import { useGitHubQuery } from "@/hooks";
 import Image from "next/image";
 import Link from "next/link";
 import { exportAsImage } from "@/utils";
+import GitHubCalendar from "react-github-calendar";
+import { useState } from "react";
+
+interface Activity {
+  date: string;
+  count: number;
+  level: 0 | 1 | 2 | 3 | 4;
+}
 
 export default function Profile() {
   const { data } = useGitHubQuery<UserProfile>(userProfileQuery);
+  const [showActivities, setShowActivities] = useState<boolean>(false);
 
   if (!data) return "Loading...";
 
+  const selectLastHalfYear = (contributions: Activity[]) => {
+    const currentYear = new Date().getFullYear();
+    const currentMonth = new Date().getMonth();
+    const shownMonths = 6;
+
+    const startDate = new Date();
+    startDate.setMonth(startDate.getMonth() - 6);
+
+    return contributions.filter((activity: Activity) => {
+      const date = new Date(activity.date);
+      const monthOfDay = date.getMonth();
+
+      return (
+        date.getFullYear() === currentYear &&
+        monthOfDay > currentMonth - shownMonths &&
+        monthOfDay <= currentMonth &&
+        startDate.getTime() <= new Date(activity.date).getTime()
```

Comment:
Now that we have `startDate.getTime() <= new Date(activity.date).getTime()` do you think we still need the other checks or are they redundant and can be removed?

Actually `date.getFullYear() === currentYear` looks wrong, if we're in february and we check for same year, only two months will be displayed as the other 4 have a different year.

Reply:
> Now that we have `startDate.getTime() <= new Date(activity.date).getTime()` do you think we still need the other checks or are they redundant and can be removed?
> 
> Actually `date.getFullYear() === currentYear` looks wrong, if we're in february and we check for same year, only two months will be displayed as the other 4 have a different year.

```suggestion
        monthOfDay > currentMonth - shownMonths &&
        monthOfDay <= currentMonth &&
        startDate.getTime() <= new Date(activity.date).getTime()
```

You're absolutely right  `date.getFullYear() === currentYear` looks wrong 
`monthOfDay > currentMonth - shownMonths &&  monthOfDay <= currentMonth` still need for get between in range Correct me if I'm wrong.

Reply:
If we want the data to be "the last full 6 months" then we need it, but if we want "the last 6 months from today (6th october -> 6th April) then the `.getTime()` check is enough.

Let's think of what design choice makes more sense, if full months or months from current date 🤔

Reply:
In my opinion maybe display months from current date, Simplify subtracts from 6 months.

![image](https://github.com/Balastrong/github-stats/assets/55313215/aba7b8da-0cdd-4fae-8e61-3c1476ed780c)

Reply:
I agree! Then the getTime() check should be enough :)